### PR TITLE
feat(common): add a version constant to each subproject

### PR DESCRIPTION
**Pull request by Laura-Danielle**
## Overview
Quick change to add a version constant to the relevant subprojects. I kept getting confused by the
number passed into the `DeviceInfoHandler`.

## Review Requests

Let me know if there is a preferred way to do this.

### CODE REVIEW

File: include/gantry/core/constants.hpp [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#pragma once
```
Comment (sotaoverride) [inline include/gantry/core/constants.hpp:1]: arent we using python scripts for generating such files with constants now?

File: head/core/can_task.cpp [hunk 2]
```diff
@@ -53,12 +54,12 @@ static auto can_move_group_handler_left =
 
 /** Handler of device info requests. */
 static auto device_info_handler_right =
-    device_info_handler::DeviceInfoHandler(right_queues, 0);
+    device_info_handler::DeviceInfoHandler(right_queues, VERSION);
```
Comment (sotaoverride) [inline head/core/can_task.cpp:57]: could prolly just use one handler to take care of both the heads.

File: include/gantry/core/constants.hpp (hunk 1 above)
Comment (amitlissack) [reply include/gantry/core/constants.hpp:1]: We are using python scripts to generate constants that are shared between C++ and Python.

**Review by amitlissack — APPROVED** [review]

**Review by sfoster1 — CHANGES_REQUESTED** [review]
We don't need to go all the way towards tying it into git, but IMO this file should be

1. in `common` somewhere
2. generated by cmake (sorry!)

**Comment by Laura-Danielle** [comment]
@amitlissack, I wonder if it makes sense for us to share a version between python and the firmware though. I could imagine a world where the python side is say v1.2, head/gantry are v1.0 and pipettes is v1.1.

However, if you think in general it's easier to keep those versions tracked on the python side I can do that.

File: head/core/can_task.cpp (hunk 2 above)
Comment (Laura-Danielle) [reply head/core/can_task.cpp:57]: I think we can address this in a follow-up PR.

**Pull request closed without merging**